autoload/airline/themes/dark.vim

```diff
@@ -1,37 +1,78 @@
-" simply overwrite the previous colors.  If you want simultaneous overrides,
-" then they will need to change different parts of the statusline so they do
-" not conflict with each other.
-"
-" First, let's define an empty dictionary and assign it to the "palette"
-" variable. The # is a separator that maps with the directory structure. If
-" you get this wrong, Vim will complain loudly.
+" For this, the dark.vim, theme, this is defined as
 let g:airline#themes#dark#palette = {}
 
-" First let's define some arrays. The s: is just a VimL thing for scoping the
-" variables to the current script. Without this, these variables would be
-" declared globally. Now let's declare some colors for normal mode and add it
-" to the dictionary.  The array is in the format:
-" [ guifg, guibg, ctermfg, ctermbg, opts ]. See "help attr-list" for valid
-" values for the "opt" value.
-let s:N1   = [ '#00005f' , '#dfff00' , 17  , 190 ]
-let s:N2   = [ '#ffffff' , '#444444' , 255 , 238 ]
-let s:N3   = [ '#9cffd3' , '#202020' , 85  , 234 ]
-let g:airline#themes#dark#palette.normal = airline#themes#generate_color_map(s:N1, s:N2, s:N3)
+" Keys in the dictionary are composed of the mode, and if specified the
+" override.  For example:
+"   * g:airline#themes#dark#palette.normal 
+"       * the colors for a statusline while in normal mode
+"   * g:airline#themes#dark#palette.normal_modified 
+"       * the colors for a statusline while in normal mode when the buffer has
+"         been modified
+"   * g:airline#themes#dark#palette.visual 
+"       * the colors for a statusline while in visual mode
+"
+" Values for each dictionary key is an array of color values that should be
+" familiar for colorscheme designers:
+"   * [guifg, guibg, ctermfg, ctermbg, opts]
+" See "help attr-list" for valid values for the "opt" value.
+"
+" Each theme must provide an array of such values for each airline section of
+" the statusline (airline_a through airline_z).  A convenience function, 
+" airline#themes#generate_color_map() exists to mirror airline_a/b/c to
+" airline_x/y/z, respectively.
+
+" The dark.vim theme:
+let s:airline_a_normal   = [ '#00005f' , '#dfff00' , 17  , 190 ]
+let s:airline_b_normal   = [ '#ffffff' , '#444444' , 255 , 238 ]
+let s:airline_c_normal   = [ '#9cffd3' , '#202020' , 85  , 234 ]
+let g:airline#themes#dark#palette.normal = airline#themes#generate_color_map(s:airline_a_normal, s:airline_b_normal, s:airline_c_normal)
+
+" It should be noted the above is equivalent to:
+" let g:airline#themes#dark#palette.normal = airline#themes#generate_color_map(
+"    \  [ '#00005f' , '#dfff00' , 17  , 190 ],  "section airline_a
+"    \  [ '#ffffff' , '#444444' , 255 , 238 ],  "section airline_b
+"    \  [ '#9cffd3' , '#202020' , 85  , 234 ]   "section airline_c
+"    \)
+"
+" In turn, that is equivalent to:
+" let g:airline#themes#dark#palette.normal = {
+"    \  [ '#00005f' , '#dfff00' , 17  , 190 ],  "section airline_a
```

Comment:
and depending on how exhaustive you want to be, it would also include all the intermediary groups, like `airline_a_to_airline_b`, etc.